src/systems/a380_systems/src/pneumatic.rs

```diff
@@ -2588,494 +2779,369 @@ mod tests {
+            .set_engine_bleed_push_button_off(1)
+            .set_engine_bleed_push_button_off(4)
+            // Close pack valves to make sure air does not bypass crossbleed valves through the packs
+            .set_pack_flow_pb_is_auto(1, false)
+            .set_pack_flow_pb_is_auto(2, false)
             .cross_bleed_valve_selector_knob(CrossBleedValveSelectorMode::Shut);
+
+        // Depressurize reservoirs
         test_bed.fail(FailureType::ReservoirAirLeak(HydraulicColor::Green));
-        test_bed.fail(FailureType::ReservoirAirLeak(HydraulicColor::Blue));
         test_bed.fail(FailureType::ReservoirAirLeak(HydraulicColor::Yellow));
         test_bed
             .test_bed
             .run_multiple_frames(Duration::from_secs(16));
 
+        assert!(test_bed.green_hydraulic_reservoir_pressure() < Pressure::new::<psi>(20.));
+        assert!(test_bed.yellow_hydraulic_reservoir_pressure() < Pressure::new::<psi>(20.));
+
+        // Make sure HP valves are closed because reservoirs should be pressurized by inboard engines system, not the HP pipe directly
+        assert!(!test_bed.hp_valve_is_open(2));
+        assert!(!test_bed.hp_valve_is_open(3));
+
         test_bed.unfail(FailureType::ReservoirAirLeak(HydraulicColor::Green));
-        test_bed.unfail(FailureType::ReservoirAirLeak(HydraulicColor::Blue));
         test_bed.unfail(FailureType::ReservoirAirLeak(HydraulicColor::Yellow));
         test_bed
             .test_bed
             .run_multiple_frames(Duration::from_secs(16));
 
-        assert!(test_bed.green_hydraulic_reservoir_pressure() > Pressure::new::<psi>(40.));
-        assert!(test_bed.blue_hydraulic_reservoir_pressure() > Pressure::new::<psi>(40.));
-        assert!(test_bed.yellow_hydraulic_reservoir_pressure() > Pressure::new::<psi>(40.));
-    }
+        // We still expect them not to be pressurized because the crossbleed valves are closed
+        assert!(test_bed.green_hydraulic_reservoir_pressure() < Pressure::new::<psi>(20.));
+        assert!(test_bed.yellow_hydraulic_reservoir_pressure() < Pressure::new::<psi>(20.));
 
-    #[test]
-    fn apu_bleed_provides_at_least_35_psi_to_left_system_with_closed_cross_bleed_valve() {
-        let test_bed = test_bed_with()
-            .stop_eng1()
-            .stop_eng2()
-            .cross_bleed_valve_selector_knob(CrossBleedValveSelectorMode::Shut)
-            .set_bleed_air_running()
-            .set_pack_flow_pb_is_auto(1, false)
-            .set_pack_flow_pb_is_auto(2, false)
+        test_bed = test_bed
+            .cross_bleed_valve_selector_knob(CrossBleedValveSelectorMode::Open)
             .and_stabilize();
 
-        assert!(!test_bed.pr_valve_is_open(1));
-        assert!(!test_bed.cross_bleed_valve_is_open());
-
-        assert!(test_bed.precooler_outlet_pressure(1) > Pressure::new::<psi>(35.));
-        assert!(!test_bed.precooler_outlet_pressure(2).is_nan());
-    }
-
-    #[test]
-    fn bleed_monitoring_computers_powered_by_correct_buses() {
```

Comment:
You don't have any tests for the `CoreProcessingInputOutputModuleA` or did I miss it?

Reply:
The logic for valve closures/regulation is tested, but that's it. This is because I have not implemented to correct relay-switching logic for CPIOM unit failures.